examples/deepspeed/gpt-neox/gpt2_trial.py

```diff
@@ -0,0 +1,274 @@
+from attrdict import AttrMap
+from datetime import datetime
+import traceback
+import numpy as np
+import torch
+import deepspeed
+import pathlib
+
+import megatron.utils as megatron_utils
+import megatron.training as megatron_train
+from megatron.data.data_utils import build_datasets_from_neox_args
+from megatron.checkpointing import save_checkpoint, load_checkpoint
+
+from determined.pytorch import DataLoader
+from determined.tensorboard.metric_writers.pytorch import TorchWriter
+from determined.pytorch.deepspeed import (
+    DeepSpeedTrial,
+    DeepSpeedTrialContext,
+)
+from determined import InvalidHP
+from det_utils import (
+    get_neox_args,
+    TensorboardWriter,
+    EarlyStoppingCallback,
+    LMReducers,
+    EvalHarness,
+)
+
+
+class GPT2Trial(DeepSpeedTrial):
+    def __init__(self, context: DeepSpeedTrialContext) -> None:
+        self.context = context
+        self.exp_config = self.context.get_experiment_config()
+        self.args = AttrMap(self.context.get_hparams())
+
+        # Initalize and get arguments, timers, and Tensorboard writer.
+        try:
+            self.neox_args = get_neox_args(self.context)
+        except:
+            traceback.print_exc()
+            raise InvalidHP("Could not parse neox_args.")
+        self.wrapped_writer = TorchWriter()
+        self.neox_args.tensorboard_writer = self.wrapped_writer.writer
+        self.neox_args.configure_distributed_args()
+        self.neox_args.build_tokenizer()  # tokenizer needs to be build in training in order to set the padding vocab
```

Comment:
question: What's this comment referencing?

Reply:
i changed the comment to clarify.  basically the tokenizer needs to be built in order to set a padded_vocab_size argument before the model is initialized.